File: SPlitwise java/src/main/java/org/example/model/Expense.java
package org.example.model;

import lombok.NonNull;

import java.util.List;

public class Expense {

    private final String id;
    private final String paidByUserID;
    private final List<Split> splits;
    private ExpenseMetadata expenseMetadata;
    private final Double amount;
    private final SplitType splitType;

    public Expense(@NonNull final String id,
                   @NonNull final String paidByUserID,
                   @NonNull final List<Split> splits,
                   @NonNull  final Double amount,
                   @NonNull final SplitType splitType) {

        this.id = id;
        this.paidByUserID = paidByUserID;
        this.splits = splits;
        this.amount = amount;
        this.splitType = splitType;
    }
}

File: Split Wise/Splitwise C#/Model.cs/Expense.cs

namespace Model
{
    public abstract class Expense
    {
        public string Id { get; set;}
        public User PaidBy { get; set;}
        public double Amount {get; set;}
        // public ExpenseType ExpenseType { get; set;}
        public List<Split> Splits {get; } = new();
        // public int NumberOfUsers { get; set;}
        public ExpenseMetadata Metadata { get; set;}
        public Expense(double amount, User paidBy, List<Split> splits, ExpenseMetadata metadata)
        {
            this.Amount = amount;
            this.PaidBy = paidBy;
            this.Splits = splits;
            this.Metadata = metadata;
        }
        public abstract bool Validate();
    }
}